### ISSUE TITLE
Add support to disable public_network_access in azurerm_bot_service_azure_bot 

### ISSUE THREAD

**Issue by DennisLundtoft** [hashicorp/terraform-provider-azurerm#23572]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Add support to disable public_network_access in azurerm_bot_service_azure_bot
azurerm_bot_service_azure_bot seem to be missing the property to disable public network access. It seems to be supported in the [SDK](https://github.com/tombuildsstuff/kermit/blob/main/sdk/botservice/2021-05-01-preview/botservice/models.go#L568)

### New or Affected Resource(s)/Data Source(s)

azurerm_bot_service_azure_bot

### Potential Terraform Configuration

```hcl
resource "azurerm_bot_service_azure_bot" "example" {
  name                = "examplebot"
  resource_group_name = var.resource_group_name
  location            = var.location
  microsoft_app_id    = var.client_id
  sku                 = "F0"

  public_network_access_enabled = false
}
```

**Comment by rcskosir** [comment]
@DennisLundtoft Thank you for taking the time to open this feature request!

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. I assume it has been supported by TF. See more details from https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/bot_channels_registration.html#public_network_access_enabled.

**Comment by jake-setford** [comment]
> Thanks for raising this issue. I assume it has been supported by TF. See more details from https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/bot_channels_registration.html#public_network_access_enabled.

@neil-yechenwei thats for the old deprecated [azurerm_bot_channels_registration](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/bot_channels_registration.html#public_network_access_enabled) @DennisLundtoft is talking about the latest offering [azurerm_bot_service_azure_bot](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/bot_service_azure_bot). I'm currently migrating a number of bots and have just hit this issue.

**Comment by drdamour** [comment]
looks like this was attempted to be added in #24125 but seems to be broken on a refresh...TF always determines the value is false for our bots

**Comment by rcskosir** [comment]
@drdamour Thank you for helping find the connecting PR. I am going to mark this issue as resolved with #24125 since the enhancement support was added with that PR. I have labeled your issue #24242 as a bug and work to fix it can be tracked via that issue. Thank you again for the help.
